db-service/lib/cqn4sql.js

```diff
@@ -1583,7 +1583,7 @@ function cqn4sql(originalQuery, model) {
         // select from anonymous query, use artificial alias
         transformedFrom.as = Object.keys(originalQuery.sources)[0]
       }
-      return { transformedFrom }
+      return { transformedFrom, transformedWhere: existingWhere }
```

Comment:
this is a "side" fix → Dont ignore where on original query if we select from a subquery.